bundles/org.openhab.binding.echonetlite/src/main/java/org/openhab/binding/echonetlite/internal/EchonetLiteHandler.java

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.echonetlite.internal;
+
+import static java.util.Objects.requireNonNull;
+import static org.openhab.binding.echonetlite.internal.EchonetLiteBindingConstants.PROPERTY_NAME_INSTANCE_KEY;
+
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EchonetLiteHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Michael Barker - Initial contribution
+ */
+@NonNullByDefault
+public class EchonetLiteHandler extends BaseThingHandler implements EchonetDeviceListener {
+    private final Logger logger = LoggerFactory.getLogger(EchonetLiteHandler.class);
+
+    private @Nullable InstanceKey instanceKey;
+    private final Map<String, State> stateByChannelId = new HashMap<>();
+
+    public EchonetLiteHandler(final Thing thing) {
+        super(thing);
+    }
+
+    @Nullable
+    private EchonetLiteBridgeHandler bridgeHandler() {
+        @Nullable
+        final Bridge bridge = getBridge();
+        if (null == bridge) {
+            logger.warn("Unable to handle commands if bridge is null");
+            return null;
+        }
+
+        @Nullable
+        final EchonetLiteBridgeHandler handler = (EchonetLiteBridgeHandler) bridge.getHandler();
+        if (null == handler) {
+            logger.warn("Unable to handle commands if bridge handler is null");
```

Comment:
This method is also called by `initialize()` which has its own handling and will also log it. This logging should probably be moved to `handleCommand()`.